src/main/java/com/freshman/freshmanbackend/domain/member/service/MemberService.java

```diff
@@ -0,0 +1,41 @@
+package com.freshman.freshmanbackend.domain.member.service;
+
+import com.freshman.freshmanbackend.domain.member.domain.Member;
+import com.freshman.freshmanbackend.domain.member.repository.MemberRepository;
+import com.freshman.freshmanbackend.domain.member.request.MemberUpdateRequest;
+import com.freshman.freshmanbackend.domain.member.response.MemberResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * 멤버 서비스
+ */
+@RequiredArgsConstructor
+@Service
+public class MemberService {
+    private final MemberRepository memberRepository;
+
+    @Transactional
+    public void save(MemberUpdateRequest request, String oauth2Id) {
+        Member member = memberRepository.findByOauth2Id(oauth2Id);
+        member.registerMember(request);
+    }
+
+    @Transactional(readOnly = true)
+    public MemberResponse get(String oauth2Id) {
+        Member member = memberRepository.findByOauth2Id(oauth2Id);
+        return MemberResponse.toResponse(member);
+    }
+
+    @Transactional
+    public void update(String oauth2Id, MemberUpdateRequest request) {
+        Member member = memberRepository.findByOauth2Id(oauth2Id);
+        member.updateMember(request);
+    }
+
+    @Transactional
+    public void delete(String oauth2Id) {
+        memberRepository.deleteByOauth2Id(oauth2Id);
+    }
```

Comment:
```suggestion
    @Transactional
    public void save(MemberUpdateRequest request) {
        Member member = memberRepository.findByOauth2Id(AuthMemberUtils.getCurrentUserOauth2Id());
        member.registerMember(request);
    }

    @Transactional(readOnly = true)
    public MemberResponse get() {
        Member member = memberRepository.findByOauth2Id(AuthMemberUtils.getCurrentUserOauth2Id());
        return MemberResponse.toResponse(member);
    }

    @Transactional
    public void update(MemberUpdateRequest request) {
        Member member = memberRepository.findByOauth2Id(AuthMemberUtils.getCurrentUserOauth2Id());
        member.updateMember(request);
    }

    @Transactional
    public void delete() {
        memberRepository.deleteByOauth2Id(AuthMemberUtils.getCurrentUserOauth2Id());
    }
```

유틸 클래스를 통해 원하는 값을 가져올 수 있다면, 필요한 곳에서만 사용하는 게 좋을 것 같습니다.
Controller 단에서 Service단에 파라미터로 oauth2Id 값을 전달해줄 필요가 있을까요??

Reply:
그렇군요! 수정했습니다!